entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -557,10 +567,184 @@ class EntityService(
+        }.mapValues {
+            JsonLdUtils.expandValueAsListOfMap(it.value)
+        }.filter {
+            isAttributeOfType(it.value[0], NGSILD_RELATIONSHIP_TYPE)
+        }.map {
+            val relationshipType = it.key.toRelationshipTypeName()
+            it.value.map {
+                val datasetIdOfRelationship = it.getDatasetId()
+                if (neo4jRepository.hasRelationshipInstance(
+                    AttributeSubjectNode(attribute.id),
+                    relationshipType,
+                    datasetIdOfRelationship
+                )
+                )
+                    partialUpdateRelationshipOfAttribute(attribute.id, relationshipType, it, datasetIdOfRelationship)
+                else 0
+            }.sum()
+        }.sum()
+    }
+
+    internal fun partialUpdatePropertiesOfAttribute(
+        attribute: Attribute,
+        attributePayload: Map<String, List<Any>>,
+        contexts: List<String>
+    ): Int {
+        return attributePayload.filterKeys {
+            !NGSILD_PROPERTIES_CORE_MEMBERS.contains(it)
+        }.mapValues {
+            JsonLdUtils.expandValueAsListOfMap(it.value)
+        }.filter {
+            isAttributeOfType(it.value[0], NGSILD_PROPERTY_TYPE)
+        }.map {
+            val expandedPropOfPropName = expandJsonLdKey(it.key, contexts)!!
+            it.value.map {
+                val datasetIdOfProperty = it.getDatasetId()
+                if (neo4jRepository.hasPropertyInstance(
+                    AttributeSubjectNode(attribute.id),
+                    expandedPropOfPropName,
+                    datasetIdOfProperty
+                )
+                )
+                    partialUpdatePropertyOfAttribute(attribute.id, expandedPropOfPropName, it, datasetIdOfProperty)
+                else 0
+            }.sum()
+        }.sum()
+    }
+
+    internal fun partialUpdateRelationshipOfAttribute(
+        attributeId: URI,
+        relationshipType: String,
+        payload: Map<String, List<Any>>,
+        datasetId: URI?
+    ): Int {
+        var updateResult = 0
+        val hasObjectEntry = payload[NGSILD_RELATIONSHIP_HAS_OBJECT]
+        val observedAt = getPropertyValueFromMapAsDateTime(payload, JsonLdUtils.NGSILD_OBSERVED_AT_PROPERTY)
+        if (hasObjectEntry != null && hasObjectEntry.isNotEmpty()) {
+            val objectId = extractRelationshipObjectOrThrowError(relationshipType, payload)
+            if (!exists(objectId))
+                throw BadRequestDataException("Target entity $objectId does not exist, create it first")
+
+            val oldRelationshipTargetId = neo4jRepository.getRelationshipTargetOfSubject(
+                attributeId,
+                relationshipType
+            )!!.id
+            updateResult += neo4jRepository.updateRelationshipTargetOfSubject(
+                attributeId,
+                relationshipType,
+                oldRelationshipTargetId,
+                objectId
+            )
```

Comment:
I think this can be done in one query (`oldRelationshipTargetId` does not seem necessary)